Sources/NIOExtras/NIORequestIdentifiable.swift

```diff
@@ -0,0 +1,19 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+public protocol NIORequestIdentifiable {
```

Comment:
It's a bit sad that `Identifiable` has platform requirements and we can't use that instead.

Reply:
Seems like I was wrong about this: is there any reason we can't use `Identifiable` instead of this?

Reply:
@glbrntt that's what I had before but @Lukasa asked me to change it to this :)

Reply:
@Lukasa what's the reason not to use `Identifiable`? 
Otherwise I'm not sure if the naming is quite right. This protocol isn't just used to identify requests but responses as well. Maybe we should name it `NIORequestReponseIdentifiable` and `var requestResponseID: RequestResponseID`? But that's quite a mouthful.

Reply:
To avoid squatting on `var id`, I believe.

Reply:
I don't really see why this is a problem as one can always create a wrapper type. But I'm fine with it if we rename it to `NIORequestResponseIdentifiable`. 
If we really don't want to decide on a name for the property we could also just let the user give the channel handler two closures in the form of `(Request) -> ID` and `(Response) -> ID` where `ID: Hashable` is an additional generic parameter of the channel handler.

Reply:
Yeah, I was reluctant for us to use `var id`, which I tend to think is over-broad. That said, I don't feel this excessively strongly, so I'm open to us just using `Identifiable` instead.

Reply:
Decided offline that we're sticking with it as is.